app/model/WeeklyRegions.scala

```diff
@@ -1,34 +1,75 @@
-}
+  private val UK = CountryGroup.UK.countries.toSet
+  private val US = CountryGroup.US.countries.toSet
+  private val ZONEC = CountryGroup.allGroups.flatMap(_.countries).toSet.diff(UK union US)
 
-case class UnitedStates(catalog: Catalog) extends WeeklyRegion {
-  override val title = "United States"
-  override val description = "Includes Alaska and Hawaii"
-  override val url = Uri.parse(s"checkout/${catalog.weeklyZoneA.quarter.slug}").addParam("countryGroup", "us")
-}
+  def all(catalog: Catalog): List[WeeklyRegion] = List(WeeklyRegion(
+    title = "United Kingdom",
+    description = "Includes Isle of Man and Channel Islands",
+    url = Uri.parse(s"checkout/${catalog.weeklyZoneA.quarter.slug}").addParam("countryGroup", "uk"),
+    ratePlans = catalog.weeklyZoneA.toList.map(_.id).toSet,
+    countries = UK
+  ), WeeklyRegion(
+    title = "United States",
+    description = "Includes Alaska and Hawaii",
+    url = Uri.parse(s"checkout/${catalog.weeklyZoneA.quarter.slug}").addParam("countryGroup", "us"),
+    ratePlans = catalog.weeklyZoneA.toList.map(_.id).toSet,
+    countries = US
+  ), WeeklyRegion(
+    title = "Rest of the world",
+    description = "Posted to you by air mail",
+    url = Uri.parse(s"checkout/${catalog.weeklyZoneC.quarter.slug}"),
+    ratePlans = catalog.weeklyZoneC.toList.map(_.id).toSet,
+    countries = ZONEC
+  ))
+
+  def forPromotion(promotion: PromoWithWeeklyLandingPage, catalog: Catalog) = {
+    val promotionProductRatePlanIds = promotion.appliesTo.productRatePlanIds
+    val promotionCountries = promotion.appliesTo.countries
+
+    all(catalog).filter { region =>
+      val ratePlaninPromotion = promotionProductRatePlanIds.exists(plan => region.ratePlans.contains(plan))
+      val countryInPromotion = !region.countries.intersect(promotionCountries).isEmpty
+      ratePlaninPromotion && countryInPromotion
+    }
+  }
+
+  private def currencyFor(country: Country): Option[Currency] = CountryGroup.availableCurrency(Currency.all.toSet)(country)
 
-case class Row(catalog: Catalog) extends WeeklyRegion {
-  override val title = "Rest of the world"
-  override val description = "Posted to you by air mail"
-  override val url = Uri.parse(s"checkout/${catalog.weeklyZoneC.quarter.slug}")
+  def prices(promotion: PromoWithWeeklyLandingPage, region: WeeklyRegion, catalog: Catalog):List[String] = {
+    def availableCountries = (region.countries intersect promotion.appliesTo.countries).toList
+
+    val cs = availableCountries.flatMap(currencyFor(_))
+
+    def plans:List[CatalogPlan.Paid] = catalog.weekly.flatten.filter(plan => promotion.appliesTo.productRatePlanIds.contains(plan.id))
+
+    for {
+      plan <- plans
+      discountPromo <- promotion.asDiscount.toList
+      currency <- plan.charges.currencies.toList
+      if (cs.contains(currency))
+    } yield {
+      plan.charges.prettyPricingForDiscountedPeriod[scalaz.Id.Id, WeeklyLandingPage](discountPromo, currency)
```

Comment:
Id.Id is even more basic and instinctual than Id alone, and should be replaced with ~the superego~ something that doesn't conflict with Option so badly. 

We have to parameterise the function call here because Id.Id is the identity type, and the scala compiler needs to be persuaded that the emperor has new clothes, otherwise it will consider the "container" type of the LandingPage to be LandingPage. 

Thanks to @philwills for figuring out the types involved in this.  (And @johnduffell for helping me remain patient when trying to figure it out to begin with)